editor/plugins/sprite_frames_editor_plugin.cpp

```diff
@@ -300,21 +327,106 @@ void SpriteFramesEditor::_sheet_zoom_reset() {
 	split_sheet_preview->set_custom_minimum_size(texture_size * sheet_zoom);
 }
 
-void SpriteFramesEditor::_sheet_select_clear_all_frames() {
-	bool should_clear = true;
+void SpriteFramesEditor::_sheet_order_selected(int p_option) {
+	frames_need_sort = true;
+	split_sheet_preview->queue_redraw();
+}
+
+void SpriteFramesEditor::_sheet_select_all_frames() {
 	for (int i = 0; i < split_sheet_h->get_value() * split_sheet_v->get_value(); i++) {
 		if (!frames_selected.has(i)) {
-			frames_selected.insert(i);
-			should_clear = false;
+			frames_selected.insert(i, selected_count++);
+			frames_need_sort = true;
 		}
 	}
-	if (should_clear) {
-		frames_selected.clear();
-	}
 
 	split_sheet_preview->queue_redraw();
 }
 
+void SpriteFramesEditor::_sheet_clear_all_frames() {
+	frames_selected.clear();
+	selected_count = 0;
+
+	split_sheet_preview->queue_redraw();
+}
+
+void SpriteFramesEditor::_sheet_sort_frames() {
+	if (!frames_need_sort) {
+		return;
+	}
+
+	frames_need_sort = false;
+
+	frames_ordered.resize(frames_selected.size());
+
+	if (frames_selected.is_empty()) {
+		return;
+	}
+
+	const Size2i frame_count = _get_frame_count();
+
+	const int frame_order = split_sheet_order->get_selected_id();
+
+	int index = 0;
+
+	// Fill based on order.
```

Comment:
Nit: There are too many empty lines here, it will probably look cleaner with some light grouping.